validator/keymanager/remote-web3signer/keymanager.go

```diff
@@ -38,54 +53,323 @@ type SetupConfig struct {
 	// Either URL or keylist must be set.
 	// a static list of public keys to be passed by the user to determine what accounts should sign.
 	// This will provide a layer of safety against slashing if the web3signer is shared across validators.
-	ProvidedPublicKeys [][48]byte
+	ProvidedPublicKeys []string
 }
 
 // Keymanager defines the web3signer keymanager.
 type Keymanager struct {
 	client                internal.HttpSignerClient
 	genesisValidatorsRoot []byte
-	publicKeysURL         string
-	providedPublicKeys    [][48]byte
+	providedPublicKeys    [][48]byte          // source of truth flag loaded + file loaded + api loaded keys
+	flagLoadedKeysMap     map[string][48]byte // stores what was provided from flag as a map
```

Comment:
I would remove the comment as it doesn't provide any extra information. The field name and type provide this info already.

Reply:
I feel like these are still helpful, the file is also loaded from flag but I can clarify this.